testing/demo/pkg/setup/groth16.go

```diff
@@ -4,70 +4,74 @@ import (
-	fmt.Println("Creating the Groth16 light client on simapp...")
-
-	ethClient, err := ethclient.Dial("http://localhost:8545")
+// CreateGroth16LightClient creates the Groth16 light client on simapp.
+func CreateGroth16LightClient() error {
+	clientCtx, err := utils.SetupClientContext()
 	if err != nil {
-		return fmt.Errorf("failed to connect to Ethereum client: %v", err)
+		return fmt.Errorf("failed to setup client context: %v", err)
 	}
-
-	genesisBlock, latestBlock, err := getGenesisAndLatestBlock(ethClient)
+	clientState, consensusState, err := createClientAndConsensusState()
 	if err != nil {
 		return err
 	}
 
-	clientState, consensusState, err := createClientAndConsensusState(genesisBlock, latestBlock)
+	fmt.Println("Creating the Groth16 light client on simapp...")
+	createClientMsgResponse, err := utils.BroadcastMessages(clientCtx, relayer, 200_000, &clienttypes.MsgCreateClient{
+		ClientState:    clientState,
+		ConsensusState: consensusState,
+		Signer:         relayer,
+	})
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to create Groth16 light client on simapp: %v", err)
 	}
-
-	clientCtx, err := utils.SetupClientContext()
-	if err != nil {
-		return fmt.Errorf("failed to setup client context: %v", err)
+	if createClientMsgResponse.Code != 0 {
+		return fmt.Errorf("failed to create Groth16 light client on simapp: %v", createClientMsgResponse.RawLog)
 	}
 
-	clientId, err := createClientOnSimapp(clientCtx, clientState, consensusState)
+	clientId, err := parseClientIDFromEvents(createClientMsgResponse.Events)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to parse client id from events: %v", err)
 	}
 	fmt.Printf("Created Groth16 light client on simapp with clientId %v.\n", clientId)
 
 	return nil
 }
 
-func createClientAndConsensusState(genesisBlock, latestBlock *ethtypes.Block) (*cdctypes.Any, *cdctypes.Any, error) {
-	// Query the info endpoint for the state transition verifier key
-	conn, err := grpc.NewClient("localhost:50052", grpc.WithTransportCredentials(insecure.NewCredentials()))
+func createClientAndConsensusState() (*cdctypes.Any, *cdctypes.Any, error) {
+	ethClient, err := ethclient.Dial(ethereumRPC)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to connect to prover: %w", err)
+		return nil, nil, fmt.Errorf("failed to connect to Ethereum client: %v", err)
 	}
-	defer conn.Close()
 
-	// TODO: Query this from the EVM rollup.
-	codeCommitment := []byte{}
+	genesisBlock, latestBlock, err := getGenesisAndLatestBlock(ethClient)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// TODO: Query the stateTransitionVerifierKey and stateMembershipVerifierKey from the EVM prover.
+	// Query the celestia prover info endpoint for the state transition verifier key
+	// conn, err := grpc.NewClient(evmProverRPC, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	// if err != nil {
+	// 	return nil, nil, fmt.Errorf("failed to connect to prover: %w", err)
+	// }
+	// defer conn.Close()
```

Comment:
How come this was commented out, don't we need the verifier keys

Reply:
Yes the Groth16 client needs the verifier keys but it was never set up with them. I started implementing in this PR but it's not necessary for step 3. This only became possible to implement recently b/c now there's an EVM-prover Docker service.

A TODO after this merges is to uncomment this code.